### PR TITLE
[TR] PIM-5006 : behat to test API key generation

### PR DESCRIPTION
**Pull request by jmleroux**
| Q                 | A
| ----------------- | ---
| Specs             |n
| Behats            |y
| Blue CI           |y
| Changelog updated |y
| Review and 2 GTM  |2
| Micro Demo (PO)   |n
| Migration script  |n
| Tech Doc          |n

### CODE REVIEW

File: features/Context/Page/User/Profile.php [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    public function getApiKey()
+    {
+        $generateButton = $this->find('css', '#btn-apigen');
+        $apiKey         = null;
+
+        if ($generateButton) {
+            $apiKey = $generateButton->getParent()->find('css', 'label')->getHtml();
+        }
+
+        return $apiKey;
```
Comment (phaseinducer) [inline features/Context/Page/User/Profile.php]: Maybe throw an exception "No API key found in the current page" ?

File: features/user/edit_my_account.feature [hunk 2]
```diff
@@ -15,3 +15,10 @@ Feature: Change my profile picture
     And I save the user
     Then I should see "User saved"
     And I should not see the default avatar
+
+  @javascript
```
Comment (phaseinducer) [inline features/user/edit_my_account.feature]: useless tag, same for the scenario above

File: features/Context/Page/User/Profile.php [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+ * @author    JM Leroux <jean-marie.leroux@akeneo.com>
+ * @copyright 2015 Akeneo SAS (http://www.akeneo.com)
+ * @license   http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ */
+class Profile extends Base
+{
+    /** @var string */
+    protected $path = '/user/profile/view';
+
+    public function getApiKey()
```
Comment (willy-ahva) [inline features/Context/Page/User/Profile.php]: doc

File: features/Context/Page/User/Profile.php [hunk 4]
```diff
@@ -0,0 +1,30 @@
+<?php
+
+namespace Context\Page\User;
+
+use Context\Page\Base\Base;
+
+/**
+ * Channel index page
```
Comment (willy-ahva) [inline features/Context/Page/User/Profile.php]: nope

File: features/Context/WebUser.php [hunk 5]
```diff
@@ -2915,4 +2915,22 @@ public function iShouldSeeTheStatusSwitcherButton($not)
             }
         }
     }
+
+    /**
+     * @Then /^The API key should (not )?be (.+)$/
```
Comment (phaseinducer) [inline features/Context/WebUser.php]: missing params

File: composer.json [hunk 6]
```diff
@@ -51,7 +51,7 @@
         "monolog/monolog": "1.10.0",
         "nelmio/api-doc-bundle": "2.9.0",
         "oneup/flysystem-bundle": "1.1.0",
-        "oro/platform": "0.10.4",
+        "oro/platform": "dev-PIM-5006-apigen",
```
Comment (sumbobyboys) [inline composer.json]: don't forget to revert :)

**Comment by sumbobyboys** [comment]
GTM

**Comment by phaseinducer** [comment]
GTM after changelog updated

**Pull request merged**